doc/e2e-tests.md

```diff
@@ -0,0 +1,215 @@
+# End to end test documentation
+
+Running e2e ensures various parts of the Compliance Operator are working correctly in a real cluster environment. And we are able to test out new rules and new features using e2e test.
+The tests file directories for e2e are under tests/e2e/
+
+`tests/e2e/e2e-tests.md` - contains all the tests we have, we can add new tests to that file.
+
+## How to add a new test
+
+First, the structure for the test is following:
+
+```go
+	testExecution{
+			Name:       "TestKubeletConfigRemediation",
+			IsParallel: false,
+			TestFn: func(t *testing.T, f *framework.Framework, ctx *framework.Context, mcTctx *mcTestCtx, namespace string) error {
+            ...
+            }
+    }
+```
+We can set `IsParallel` to instruct the test run as Parallel or Serial.
```

Comment:
I think it would be nice to give some best practices as to when can you use parallel or serial. My rule of thumb has been to set 'read-only' tests as parallel and anything that modifies the cluster state as serial.

Reply:
Sounds good! I will add that